lib/ASRouterTargeting.jsm

```diff
@@ -302,15 +314,17 @@ this.ASRouterTargeting = {
     OTHER_ERROR: "OTHER_ERROR",
   },
 
-  isMatch(filterExpression, customContext) {
-    let context = this.Environment;
-    if (customContext) {
-      context = {};
-      Object.defineProperties(context, Object.getOwnPropertyDescriptors(this.Environment));
-      Object.defineProperties(context, Object.getOwnPropertyDescriptors(customContext));
-    }
+  // Combines the getter properties of two objects without evaluating them
+  combineContexts(contextA = {}, contextB = {}) {
+    const context = {};
+    Object.defineProperties(context, Object.getOwnPropertyDescriptors(contextA));
```

Comment:
I wonder if we want to explicitly check for/ensure we are not overwriting properties at some point (or maybe we want to allow that?)

Reply:
Added a `reportError` call if that happens.